docs/TuxReact.md

```diff
@@ -0,0 +1,449 @@
+4. See [here](http://facebook.github.io/react/docs/jsx-gotchas.html) for all other JSX gotchas.
+
+***
+
+#### <a id="React-Components-Accessing-the-DOM-from-React"></a>3.2) Accessing the DOM from React [#](#React-Components-Accessing-the-DOM-from-React)
+`React` provides the `getDOMNode` method for accessing the actual DOM node for a particular component. This can be valuable for interacting with other libraries such as `jQuery` or reading from inputs. Normally, to use `getDOMNode` you will need to pass in a `ref` property to the component. The `ref` property allows the owning component to access the component via `this.refs.REF`. **NOTE** you cannot access the `ref` property from within the ownee component itself.
+
+```
+    var InputForm = React.createClass({
+      handleSubmit: function (e) {
+        e.preventDefault(),
+        this.addTodo(this.refs.inputField.value);
+      },
+
+      render: function () {
+        <form onSubmit={this.handleSubmit}>
+          <input ref='inputField' />
+        </form>
+      }
+    });
+```
+
+**NOTE** here that we can access the `form` `submit` event by passing in a function under the `onSubmit` property. This is how we hook into any `DOM` event: `onEventName`.
+
+***
+
+#### <a id="React-Components-Props-and-State"></a>3.3) Props and State [#](#React-Components-Props-and-State)
+In `React` components have two ways of defining changing data, `props` and `state`. `Props` are passed down from parent components and are never updated by the component itself (under almost all circumstances). `state` is managed within the component itself and used for things like getting data from models and stores or third party APIs. It is also used for [two-way data binding in form/input validation](http://facebook.github.io/react/docs/two-way-binding-helpers.html). You never modify the `state` object directly but instead use methods like `this.setState` and `this.replaceState` to update the `state` in the component.
+
+Other than in the case of input validation `React` follows a one-way data flow in which very few top level components have `state` and they pass down `props` based on that state to their child components. Those components pass down `props` to their children and so on and so forth. There is a great doc [here](http://facebook.github.io/react/docs/thinking-in-react.html) about how to think in `React` and this one-way flow of data. `Props` should be things like application data, but they should also be the methods to manipulate that data. For example, if you want your `Todo` components to be able to delete themselves, they should have their delete method passed in via `props`. Why is this? Because it allows your components to be as reusable as possible. Additionally, it reduces the number of components that have to know about your actions, ajax-requests, etc. Here is what this might look like in an example `state` managing top level component
```

Comment:
Recommend changing `Props` to `props` at the start of one of the middle sentences.